content/en/docs/v3.6/dev-internal/discovery_protocol.md

```diff
@@ -33,85 +36,72 @@ UUID=$(uuidgen)
+etcdctl --endpoints=http://example.com:2379 put /_etcd/registry/${UUID}/_config/size ${cluster_size}
 ```
 
 Usually the cluster size is 3, 5 or 7. Check [optimal cluster size][cluster-size] for more details.
 
 ### Bringing up etcd processes
 
-Given the discovery URL, use it as `-discovery` flag and bring up etcd processes. Every etcd process will follow this next few steps internally if given a `-discovery` flag.
-
-### Registering itself
+Set the discovery token `${UUID}` to `--discovery-token` flag, and set the endpoints of the etcd cluster backing the discovery service to `--discovery-endpoints` flag,  then v3 discovery is enabled to bootstrap the etcd cluster.
 
-The first thing for etcd process is to register itself into the discovery URL as a member. This is done by creating member ID as a key in the discovery URL.
+Every etcd process will follow the next few steps internally if given `--discovery-token` and `--discovery-endpoints` flags.
 
+If the discovery service enables client cert authentication, configure the following flags appropriately as well. They follow exactly the same usage as `etcdctl` communicating with an etcd cluster.
 ```
-curl -X PUT http://example.com/v2/keys/_etcd/registry/${UUID}/${member_id}?prevExist=false -d value="${member_name}=${member_peer_url_1}&${member_name}=${member_peer_url_2}"
+--discovery-insecure-transport
+--discovery-insecure-skip-tls-verify
+--discovery-cert
+--discovery-key
+--discovery-cacert
 ```
 
-### Checking the status
-
-It checks the expected cluster size and registration status in discovery URL, and decides what the next action is.
-
+If the discovery service enables role based authentication, configure the following flags appropriately as well. They follow exactly the same usage as `etcdctl` communicating with an etcd cluster.
 ```
-curl -X GET http://example.com/v2/keys/_etcd/registry/${UUID}/_config/size
-curl -X GET http://example.com/v2/keys/_etcd/registry/${UUID}
+--discovery-user
+--discovery-password
 ```
 
-If registered members are still not enough, it will wait for left members to appear.
-
-If the number of registered members is bigger than the expected size N, it treats the first N registered members as the member list for the cluster. If the member itself is in the member list, the discovery procedure succeeds and it fetches all peers through the member list. If it is not in the member list, the discovery procedure finishes with the failure that the cluster has been full.
-
-In etcd implementation, the member may check the cluster status even before registering itself. So it could fail quickly if the cluster has been full.
-
-### Waiting for all members
-
-The wait process is described in detail in the [etcd API documentation][api].
-
+The default time or timeout values can also be changed using the following flags, which also follow exactly the same usage as `etcdctl` communicating with an etcd cluster.
```

Comment:
Same as above for the second half of the sentence.

Reply:
Thanks & updated.